components/application-tag-mgt/org.wso2.carbon.identity.application.tag.mgt/src/main/java/org/wso2/carbon/identity/application/tag/mgt/dao/impl/ApplicationTagDAOImpl.java

```diff
@@ -0,0 +1,256 @@
+                                ApplicationTagManagementConstants.ErrorMessages.ERROR_CODE_APP_TAG_ALREADY_EXISTS,
+                                String.valueOf(tenantId));
+                    }
+                }
+                throw e;
+            }
+        } catch (SQLException e) {
+            throw ApplicationTagManagementUtil.handleServerException(
+                    ApplicationTagManagementConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_ADDING_APP_TAG, e);
+        }
+    }
+
+    @Override
+    public List<ApplicationTagsListItem> getAllApplicationTags(String tenantDomain) throws ApplicationTagMgtException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (log.isDebugEnabled()) {
+            log.debug("Reading all Application Tags of Tenant " + tenantId);
+        }
+        List<ApplicationTagsListItem> applicationTagsList = new ArrayList<>();
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement preparedStatement = dbConnection.prepareStatement(SQLConstants.GET_ALL_APP_TAGS)) {
+            preparedStatement.setInt(1, tenantId);
+
+            ResultSet applicationTagsResultSet = preparedStatement.executeQuery();
+
+            while (applicationTagsResultSet.next()) {
+                ApplicationTagsListItem.ApplicationTagsListItemBuilder appTagBuilder =
+                        new ApplicationTagsListItem.ApplicationTagsListItemBuilder()
+                                .id(applicationTagsResultSet.getString(SQLConstants.APP_TAG_ID_COLUMN_NAME))
+                                .name(applicationTagsResultSet.getString(SQLConstants.APP_TAG_NAME_COLUMN_NAME))
+                                .colour(applicationTagsResultSet.getString(SQLConstants.APP_TAG_COLOUR_COLUMN_NAME));
+                applicationTagsList.add(appTagBuilder.build());
+            }
+        } catch (SQLException e) {
+            throw ApplicationTagManagementUtil.handleServerException(
+                    ApplicationTagManagementConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_RETRIEVING_APP_TAGS, e);
+        }
+        return applicationTagsList;
+    }
+
+    @Override
+    public ApplicationTagsListItem getApplicationTagById(String applicationTagId, String tenantDomain)
+            throws ApplicationTagMgtException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (log.isDebugEnabled()) {
+            log.debug("Reading Application Tag " + applicationTagId);
+        }
+        ApplicationTagsListItem applicationTag = null;
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement preparedStatement = dbConnection.prepareStatement(SQLConstants.GET_APP_TAG_BY_ID)) {
+            preparedStatement.setString(1, applicationTagId);
+            preparedStatement.setInt(2, tenantId);
+
+            ResultSet resultSet = preparedStatement.executeQuery();
```

Comment:
ResultSet is not closed.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/4904/commits/3d8ec5b161358637ba2c96ed4155edb1ccd619dc